## Analýza zamestnancov VŠ

In [1]:
import pandas as pd

from pandasql import sqldf
# from sklearn import datasets

In [2]:
df = pd.read_csv('../data/preprocessed/regzam.csv', index_col=0)
display(df)

Employee_ID GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690    Štefan       Kočan                      Doc.   
1      14002860472     Dušan       Korgo                       DOC   
2      17922733804       Ján     Buzalka                     prof.   
3      16290288493     Jozef   Stieranka                     prof.   
4      21401303068     Ľuboš       Wäldl                      doc.   
...            ...       ...         ...                       ...   
32725  12110964830      Joža      Spurný                      doc.   
32726  13500390262   Marcela   Harčárová                       NaN   
32727  17108478442      Jana    Juhásová                       NaN   
32728  15917135755   Miriama    Kubušová                       NaN   
32729  51680829014      Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
1                                                    NaN               NaN   
2                                                    NaN               NaN   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
1                  NaN                                                NaN   
2                  NaN                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                                           

In [3]:
id_set = set(df["Employee_ID"])
print(len(id_set))

32730


-> máme Dataset s 32730 riadkami, pričom žiadne 2 Employee_ID nie sú rovnaké

In [4]:
name_set = set(df["FamilyName"] + " " + df["GivenName"])
len(name_set)

31639

-> 31639 rôznych mien; zostáva zistiť, či pri duplikátnych menách sa jedná o tú istú osobu, alebo nie.
Tiež by nás mohlo zaujímať, ako ukladali osoby, ktorým sa zmenilo meno...

In [5]:
def find_name_duplicates(df):
    name_set = set()
    name_duplicates = set()
    duplicates_ids = set()
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_set:
            name_duplicates.add(name)
        else:
            name_set.add(name)
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_duplicates:
            duplicates_ids.add(row["Employee_ID"])
    return tuple(name_duplicates), tuple(duplicates_ids)

In [6]:
name_duplicates, ids = find_name_duplicates(df)

In [7]:
q = f"SELECT * FROM df WHERE Employee_ID IN {ids}"
duplicates_df = sqldf(q, globals()).sort_values(by=['FamilyName', 'GivenName'])
display(duplicates_df)

Employee_ID  GivenName FamilyName DegreePedagogicalAcademic  \
1327  49910822214   Katarína  Adamíková                      None   
1576  18913486096   Katarína  Adamíková                      None   
114   39501978150  František    Adamčík                     prof.   
1325  17240205086  František    Adamčík                      None   
133   51642863772       Jana   Antalová                      None   
...           ...        ...        ...                       ...   
1876  15552653387      Tomáš      Žilka                      None   
1107  10056354153   Viktória    Žilková                      None   
1994  14058627798   Viktória    Žilková                      None   
530   14639882132       Anna    Žuffová                      None   
1296  96699806419       Anna    Žuffová                      None   

     DegreeMaster DegreeDoctor  \
1327         Mgr.         None   
1576         Mgr.         None   
114          Ing.         CSc.   
1325         Ing.         PhD.   
133         MUDr.         PhD.   
...           ...          ...   
1876        MUDr.         None   
1107         Mgr.         None   
1994         None         MSc.   
530          None         None   
1296         Mgr.         None   

                     Employment_Workplace_University  \
1327                                            None   
1576                                            None   
114                 Technická univerzita v Košiciach   
1325                                            None   
133               Univerzita Komenského v Bratislave   
...                                              ...   
1876  Slovenská zdravotnícka univerzita v Bratislave   
1107                                            None   
1994              Univerzita Komenského v Bratislave   
530                                             None   
1296                                            None   

                                        Employment_Type  \
1327                                               None   
1576                                               None   
114                                            Profesor   
1325                                               None   
133                                    Odborný asistent   
...                                                 ...   
1876  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
1107                                               None   
1994                     Výskumný pracovník - výskumník   
530                                                None   
1296                                               None   

                                     Employment_Subtype Employment_BeginDate  \
1327                                               None                 None   
1576                                               None                 None   
114   s titulom profesor a vysokoškolským vzdelaním ...           2005-02-01   
1325                                               None                 None   
133          s vysokoškolským vzdelaním tretieho stupňa           2020-08-15   
...                                                 ...                  ...   
1876          s vysokoškolským vzdelaním druhého stupňa           2022-02-01   
1107                                               None                 None   
1994          s vysokoškolským vzdelaním druhého stupňa           2022-06-15   
530                                                None                 None   
1296                                               None                 None   

     Employment_Ratio                             Guarantee_StudyProgram  \
1327             None                                               None   
1576             None                                               None   
114               100  ['prevádzka lietadiel', 'prevádzka lietadiel',...   
1325             None                                               None   
133               100                        

In [8]:
df.query("GivenName == 'Viktória' and FamilyName == 'Žilková'")

Employee_ID GivenName FamilyName DegreePedagogicalAcademic  \
17361  10056354153  Viktória    Žilková                       NaN   
32149  14058627798  Viktória    Žilková                       NaN   

      DegreeMaster DegreeDoctor     Employment_Workplace_University  \
17361         Mgr.          NaN                                 NaN   
32149          NaN         MSc.  Univerzita Komenského v Bratislave   

                      Employment_Type  \
17361                             NaN   
32149  Výskumný pracovník - výskumník   

                              Employment_Subtype Employment_BeginDate  \
17361                                        NaN                  NaN   
32149  s vysokoškolským vzdelaním druhého stupňa           2022-06-15   

      Employment_Ratio Guarantee_StudyProgram Guarantee_University  \
17361              NaN                    NaN                  NaN   
32149               50                    NaN                  NaN   

      Guarantee_Faculty Guarantee_FormOfStudy Guarantee_LevelOfStudy  \
17361               NaN                   NaN                    NaN   
32149               NaN                   NaN                    NaN   

      Employment_Workplace_Faculty Employment_EndDate  
17361                          NaN                NaN  
32149             Lekárska fakulta                NaN

... V tomto bode sme zistili, že naše stiahnuté xml zrejme obsahuje len dáta pre aktuálnych zamestnancov...

Pre minulých (zamestnancov) neobsahuje údaje o zamestnaní, predovšetkým nevieme z dát zistiť, v akom období a kde boli zamestnaní, hoci ich meno je v dátach.

In [9]:
df.query("Employment_EndDate.isna()")

Employee_ID GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690    Štefan       Kočan                      Doc.   
1      14002860472     Dušan       Korgo                       DOC   
2      17922733804       Ján     Buzalka                     prof.   
3      16290288493     Jozef   Stieranka                     prof.   
4      21401303068     Ľuboš       Wäldl                      doc.   
...            ...       ...         ...                       ...   
32725  12110964830      Joža      Spurný                      doc.   
32726  13500390262   Marcela   Harčárová                       NaN   
32727  17108478442      Jana    Juhásová                       NaN   
32728  15917135755   Miriama    Kubušová                       NaN   
32729  51680829014      Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
1                                                    NaN               NaN   
2                                                    NaN               NaN   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
1                  NaN                                                NaN   
2                  NaN                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                                           

## -> Až 32434 riadkov nemá EndDate, čo aj potvrdzuje našu hypotézu, že máme zlé dáta :(  (iba súčasné úväzky)

## **- Pracujú zamestnanci aj na viacerých pracoviskách a vidno v tom nejaké trendy alebo zhluky?**

In [112]:
df_workplaces = pd.read_csv('../data/preprocessed/regzam_workplaces_current.csv', index_col=0)
display(df_workplaces)

Employee_ID  FamilyName     GivenName DegreePedagogicalAcademic  \
0      18799121690       Kočan        Štefan                      Doc.   
1      16290288493   Stieranka         Jozef                     prof.   
2      21401303068       Wäldl         Ľuboš                      doc.   
3      69828748660     Kostrec         Matej                       NaN   
4      91221451321  Krajníková     Magdaléna                       NaN   
...            ...         ...           ...                       ...   
14326  17628642100       Peter          Fabo                       NaN   
14327  17628642100       Peter          Fabo                       NaN   
14328  28653015119    Marianna  Berinšterová                       NaN   
14329  28653015119    Marianna  Berinšterová                       NaN   
14330  28653015119    Marianna  Berinšterová                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             Ing.         PhD.   
2            JUDr.         PhD.   
3            JUDr.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
14326        RNDr.         PhD.   
14327        RNDr.         PhD.   
14328         Mgr.         PhD.   
14329         Mgr.         PhD.   
14330         Mgr.         PhD.   

                         Employment_Workplace_University  \
0      Vysoká škola bezpečnostného manažérstva v Koši...   
1                             Akadémia Policajného zboru   
2                             Akadémia Policajného zboru   
3                             Akadémia Policajného zboru   
4                             Akadémia Policajného zboru   
...                                                  ...   
14326                       Žilinská univerzita v Žiline   
14327  Trenčianska univerzita Alexandra Dubčeka v Tre...   
14328                     Prešovská univerzita v Prešove   
14329  Vysoká škola medzinárodného podnikania ISM Slo...   
14330                     Prešovská univerzita v Prešove   

                                         Employment_Type  Employment_Ratio  
0                                                 Docent            100.00  
1                                               Profesor            100.00  
2                                               Profesor            100.00  
3                                       Odborný asistent            100.00  
4                                       Odborný asistent            100.00  
...                                                  ...               ...  
14326                     Výskumný pracovník - výskumník             20.00  
14327                     Výskumný pracovník - výskumník            100.00  
14328                                   Odborný asistent            100.00  
14329  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...              5.00  
14330  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...             15.55  

[14331 rows x 9 columns]

In [114]:
# kontrola - Matusov skolitel :D :
df_workplaces[df_workplaces['FamilyName'] == 'Výrost']

Employee_ID FamilyName GivenName DegreePedagogicalAcademic DegreeMaster  \
346  53060727347     Výrost     Tomáš                      doc.         Ing.   
347  53060727347     Výrost     Tomáš                      doc.         Ing.   

    DegreeDoctor     Employment_Workplace_University Employment_Type  \
346         PhD.  Ekonomická univerzita v Bratislave          Docent   
347         PhD.    Technická univerzita v Košiciach          Docent   

     Employment_Ratio  
346             100.0  
347              50.0